docs/endpoints.md

```diff
@@ -82,6 +82,100 @@ Each signature response contains the following fields:
+
+-   **keyid**: allows the caller to specify a key to sign the data with.
+    This parameter is optional, and Autograph will pick a key based on
+    the caller\'s permission if omitted.
+-   **options**: a JSON object used to pass signer-specific options in
+    the request. Refer to the documentation of each signer to find out
+    which options they accept.
+-   **files**: an array of dicts of file name and base64 encoded content to sign. For example:
+
+```json
+[
+    {
+        "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+        "name": "sphinx_1.7.2-1.dsc"
+    },
+    {
+        "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+        "name": "sphinx_1.7.2-1.changes"
+    }
+]
+```
+
+The number of files must be between 1 and 32 inclusive. All characters
+in a file name must be an alphanumeric character, dash, underscore or
+dot. The file name must start with an alphanumeric character and dots
+cannot occur next to each other in the file name.
+
+example:
+
+``` bash
+POST /sign/file
+Host: autograph.example.net
+Content-type: application/json
+Authorization: Hawk id="alice", mac="756lSgQEYLoc6V0Uv2wS8pRg/h+4WFUVKWQynCFvY8Y=", ts="1524487134", nonce="MrpGL35q", hash="9m3WhtGQDuHermi5fDYBGJlOqNeK5B3nk0lKreZ+YSw=", ext="933126753"
+
+[
+    {
+      "files":[
+         {
+             "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+             "name": "sphinx_1.7.2-1.dsc"
+         },
+         {
+             "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+             "name": "sphinx_1.7.2-1.changes"
+         }
+      ]
+    },
+    {
+      "files":[
+         {
+             "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+             "name": "sphinx_1.7.2-1.dsc"
+         },
+         {
+             "content": "UEsDBBQACAAIAAAAAAAAAAAAAAAAAAAAAAATAAAAQW5kcm9pZE1hbmlmZXN0LnhtbKSYS2ybx7XHf0PqbVmW4...BwAACigAAAAA",
+             "name": "sphinx_1.7.2-1.changes"
+         },
+      ],
+      "keyid":"randompgp-debsign",
+      "options":null
+    }
+]
+```
+
+### Response
+
+A successful request returns `201 Created` with a response body
+containing all signed files encoded in JSON. The ordering of the
+response array is identical to the request array, such that signing
+request 0 maps to signing response 0, etc.
+
+The response format is the same as `/sign/data` except
+instead of the `signature` field autograph returns the
+field:
+
+-   `signed_files` an array of dicts of file name and base64 encoded
+    signed file content as described in the request section. Each
+    signer uses a different format, so refer to their documentation
+    for more information.
+
+TODO: document restrictions
```

Comment:
What "restrictions" does this refer to?

Reply:
Ah, this was a stray stash. Removed in 84ed7180

The filename restrictions are mentioned in [the /sign/files request body section](https://github.com/mozilla-services/autograph/pull/782/files#diff-b52944c3b61aec977284cbe459a436d41c0ee9193582df47cfe8ecad1c29066dR116-R120).